### ISSUE TITLE
Crash when clicking in certain options in the "App Ops" tab

### ISSUE THREAD

**Issue by RickyM7** [MuntashirAkon/AppManager#201]
**Describe the bug**
Clicking the 'Deny dangerous app' ops and ~~'Set custom app op'~~ options causes crashes.

**To Reproduce**
Steps to reproduce the behaviour:
1. Go to 'App Ops' tab in any app
2. Click on 'Deny dangerous app ops' or ~~'Set custom app op'~~
3. See error

**Crash logs**
**In "Deny dangerous app ops"**:

java.lang.RuntimeException: Couldn't connect to appOpsService locally
    at io.github.muntashirakon.AppManager.appops.AppOpsService.<init>(AppOpsService.java:45)
    at io.github.muntashirakon.AppManager.details.AppDetailsViewModel.ignoreDangerousAppOps(AppDetailsViewModel.java:452)
    at io.github.muntashirakon.AppManager.details.AppDetailsFragment.lambda$onOptionsItemSelected$14$AppDetailsFragment(AppDetailsFragment.java:363)
    at io.github.muntashirakon.AppManager.details.-$$Lambda$AppDetailsFragment$qrZbw68KsLdlAVvjaxI4dQch348.run(lambda)
    at java.lang.Thread.run(Thread.java:760)

Device Info:
App version: 2.5.21-DEBUG
App version code: 376
Android build version: V11.0.2.0.NCFMIXM
Android release version: 7.0
Android SDK version: 24
Android build ID: AL1512-mido-build-20191107001817
Device brand: xiaomi
Device manufacturer: Xiaomi
Device name: mido
Device model: Redmi Note 4
Device product name: mido
Device hardware name: qcom
ABIs: [arm64-v8a, armeabi-v7a, armeabi]
ABIs (32bit): [armeabi-v7a, armeabi]
ABIs (64bit): [arm64-v8a]
System language: pt-BR
In-App Language: auto

**In "Set custom app op"**:

java.lang.ArrayIndexOutOfBoundsException: length=32; index=32
    at io.github.muntashirakon.AppManager.appops.AppOpsManager.opToName(AppOpsManager.java:1482)
    at io.github.muntashirakon.AppManager.details.AppDetailsFragment.getAppOpNames(AppDetailsFragment.java:561)
    at io.github.muntashirakon.AppManager.details.AppDetailsFragment.onOptionsItemSelected(AppDetailsFragment.java:378)
    at androidx.fragment.app.Fragment.performOptionsItemSelected(Fragment.java:3094)
    at androidx.fragment.app.FragmentManager.dispatchOptionsItemSelected(FragmentManager.java:3207)
    at androidx.fragment.app.FragmentController.dispatchOptionsItemSelected(FragmentController.java:416)
    at androidx.fragment.app.FragmentActivity.onMenuItemSelected(FragmentActivity.java:346)
    at androidx.appcompat.app.AppCompatActivity.onMenuItemSelected(AppCompatActivity.java:264)
    at androidx.appcompat.view.WindowCallbackWrapper.onMenuItemSelected(WindowCallbackWrapper.java:109)
    at androidx.appcompat.view.WindowCallbackWrapper.onMenuItemSelected(WindowCallbackWrapper.java:109)
    at androidx.appcompat.app.ToolbarActionBar$2.onMenuItemClick(ToolbarActionBar.java:65)
    at androidx.appcompat.widget.Toolbar$1.onMenuItemClick(Toolbar.java:207)
    at androidx.appcompat.widget.ActionMenuView$MenuBuilderCallback.onMenuItemSelected(ActionMenuView.java:779)
    at androidx.appcompat.view.menu.MenuBuilder.dispatchMenuItemSelected(MenuBuilder.java:834)
    at androidx.appcompat.view.menu.MenuItemImpl.invoke(MenuItemImpl.java:158)
    at androidx.appcompat.view.menu.MenuBuilder.performItemAction(MenuBuilder.java:985)
    at androidx.appcompat.view.menu.MenuPopup.onItemClick(MenuPopup.java:128)
    at android.widget.AdapterView.performItemClick(AdapterView.java:310)
    at android.widget.AbsListView.performItemClick(AbsListView.java:1161)
    at android.widget.AbsListView$PerformClick.run(AbsListView.java:3142)
    at android.widget.AbsListView$3.run(AbsListView.java:4057)
    at android.os.Handler.handleCallback(Handler.java:754)
    at android.os.Handler.dispatchMessage(Handler.java:95)
    at android.os.Looper.loop(Looper.java:165)
    at android.app.ActivityThread.main(ActivityThread.java:6375)
    at java.lang.reflect.Method.invoke(Native Method)
    at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:912)
    at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:802)

Device Info:
App version: 2.5.21-DEBUG
App version code: 376
Android build version: V11.0.2.0.NCFMIXM
Android release version: 7.0
Android SDK version: 24
Android build ID: AL1512-mido-build-20191107001817
Device brand: xiaomi
Device manufacturer: Xiaomi
Device name: mido
Device model: Redmi Note 4
Device product name: mido
Device hardware name: qcom
ABIs: [arm64-v8a, armeabi-v7a, armeabi]
ABIs (32bit): [armeabi-v7a, armeabi]
ABIs (64bit): [arm64-v8a]
System language: pt-BR
In-App Language: auto

**Device info**
 - Device: Redmi Note 4
 - OS Version: MIUI 11 (Android 7.0)
 - App Manager Version: v2.5.21-DEBUG#627/#628
 - Mode: no-root

**Comment by MuntashirAkon** [comment]
Second crash should be fixed in 564c9edf

**Comment by RickyM7** [comment]
@MuntashirAkon Yes, it's been fixed. Thank you.

**Comment by MuntashirAkon** [comment]
First crash should be fixed in 18565f4f.

**Comment by RickyM7** [comment]
Fixed. I'll close the issue now.

**Comment by MuntashirAkon** [comment]
@RickyM7 a bit off-topic, but please check if all no-root features are working correctly as I've changed the sources quite a bit and don't have any old devices to test the changes. Thanks.

**Comment by RickyM7** [comment]
@MuntashirAkon It's not a problem. In case I find any crash or error I will open a new issue.

**Comment by RickyM7** [comment]
@MuntashirAkon I tested both v2.5.20 and v2.5.21-DEBUG#646 and noticed a few things, but because they are present in both stable and debug versions I preferred not to open an issue for these things.
- Attempting to back up with the "Date" option checked will cause the operation to fail.
- With the exception of "Backup APK", "Backup/Restore" and "Export Blocking Rules", all other batch operations do not work in "No Root" mode. Some of these operations even show the message "Done" (such as "Uninstall" for example) but still do not work.

I tested what I could on v2.5.21-DEBUG#646 and, apart from these things listed and the reported crash I found nothing different from v2.5.20. Looks like everything's working as expected.
